airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/setup_and_teardown_runner.py

```diff
@@ -17,16 +17,15 @@
 async def _build_container(dagger_client: dagger.Client, dockerfile_path: Path) -> dagger.Container:
     workspace = (
         dagger_client.container()
-        .with_directory("/tmp", dagger_client.host().directory(os.path.dirname(dockerfile_path)))
-        .with_workdir("/tmp")
-        .with_file("/tmp/Dockerfile.cat_setup_teardown", dagger_client.host().file(str(dockerfile_path.expanduser())))
-        .directory("/tmp")
+        .with_mounted_directory("/tmp/setup_teardown_context", dagger_client.host().directory(os.path.dirname(dockerfile_path)))
+        .directory("/tmp/setup_teardown_context")
     )
     return await dagger_client.container().build(context=workspace, dockerfile="Dockerfile.cat_setup_teardown")
 
 
-async def _build_setup_container(dagger_client: dagger.Client, dockerfile_path: Path) -> dagger.Container:
-    return await _build_container(dagger_client, dockerfile_path)
+async def _build_setup_container(dagger_client: dagger.Client, connector_path: Path, dockerfile_path: Path) -> dagger.Container:
+    container = await _build_container(dagger_client, dockerfile_path)
+    return container.with_mounted_directory("/connector", dagger_client.host().directory(str(connector_path)))
```

Comment:
Arent we somewhat mounting the connector directory twice? (if the Dockerfile is inside the connector directory).

Reply:
We're mounting the Dockerfile & context to a container that uses the Dockerfile to create this setup/teardown container, and then mounting the connector code to that setup/teardown container. I don't expect the Dockerfile's context to actually have all of the connector code, but I suppose that depends on DB Sources' implementation.